server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -489,28 +489,48 @@ public void copyFrom(Directory from, String src, IOContext context, ActionListen
      * @param destinationPath The destination path (if multipart is supported)
      * @param fileCompletionListener The listener to notify of completion
      */
-    public void copyTo(String source, Directory destinationDirectory, Path destinationPath, ActionListener<String> fileCompletionListener) {
+    public void copyTo(
+        String source,
+        Directory destinationDirectory,
+        Path destinationPath,
+        DirectoryFileTransferTracker tracker,
+        ActionListener<String> fileCompletionListener
+    ) {
         final String blobName = getExistingRemoteFilename(source);
         if (destinationPath != null && remoteDataDirectory.getBlobContainer() instanceof AsyncMultiStreamBlobContainer) {
+            long length = 0L;
+            try {
+                length = fileLength(source);
+            } catch (IOException ex) {
+                logger.error("Unable to fetch segment length for stats tracking", ex);
+            }
+            final long fileLength = length;
+            final long startTime = System.currentTimeMillis();
```

Comment:
Please use nano time for measuring elapsed time

Reply:
Sure I will update this

Reply:
Let's stick with using the wall clock here in this PR since it is baked into the class already. @Rishikesh1159 can you take #10407 as a follow up to refactor after this PR?

Reply:
Sure I will take #10407 as follow up.